sky/sky_logging.py

```diff
@@ -143,3 +168,38 @@ def is_silent():
         # threads.
         _logging_config.is_silent = False
     return _logging_config.is_silent
+
+
+def _initialize_tmp_file_logger(logger_name: str) -> str:
+    """Initialize the logger to write to a tmp file."""
+    run_timestamp = datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')
+
+    # set up the logger to write to a tmp file
+    log_dir = os.path.join(constants.SKY_LOGS_DIRECTORY, run_timestamp)
+    log_path = os.path.expanduser(
+        os.path.join(log_dir,
+                     logger_name.replace('.', '_') + '.log'))
+    os.makedirs(os.path.dirname(log_path), exist_ok=True)
+    log_abs_path = pathlib.Path(log_path).expanduser().absolute()
+    fh = logging.FileHandler(log_abs_path)
+
+    logger = logging.getLogger(logger_name)
+
+    fh.setFormatter(FORMATTER)
+    fh.setLevel(_root_logger.level)
+    logger.addHandler(fh)
+    # Disable propagate to avoid streaming logs to the console
+    logger.propagate = False
+
+    return log_path
+
+
+# A map from logger name to a tuple of (initializer, is_initialized, log_path).
+# The initializer is a function that initializes the logger.
+# The is_initialized is a boolean indicating if the logger is initialized.
+_LOGGER_NAME_INITIALIZER_MAP: Dict[str,
+                                   Tuple[Callable[[str], str], bool, str]] = {
+                                       STORAGE_LOGGER_NAME:
+                                           (_initialize_tmp_file_logger, False,
+                                            '')
+                                   }
```

Comment:
Can we make this a dataclass for readability?

Reply:
Removed since we're gonna to use `log_lib.run_with_log()`